src/Components/Calculator.js

```diff
@@ -1,35 +1,22 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import Row from './RowCalcul';
 import calculate from './logic/calculator';
 import '../css/Calc.css';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      dataObj: {
-        total: null,
-        next: null,
-        operation: null,
-      },
-    };
-    this.updateDataObj = this.updateDataObj.bind(this);
-  }
-
-  componentDidUpdate() {
-    this.updateScreen();
-  }
+export default function Calculator() {
+  const [dataObj, setDataObj] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-  updateDataObj(event) {
-    const val = event.target.textContent;
-    this.setState((prevState) => ({
-      dataObj: { ...calculate(prevState.dataObj, val) },
-    }));
+  function updateDataObj(event) {
+    const value = event.target.textContent;
+    setDataObj((prevDataObj) => calculate(prevDataObj, value));
   }
```

Comment:
- [ ] Kindly use ES6 arrow function syntax to declare all of your functions in your Calculator component instead of the `function` keyword. 
``` Js
const Calculator = () => {
...
}'
```